typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,440 @@
+            defaultPartitionId,
+            containerCreationConfig,
+            generateRunConfigFunc,
+            )
+        if (err != null) {
+            return [null, null, err ] //TODO no personalized error message
+        }
+
+        return [serviceContext, hostPortBindings, null]
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public AddServiceToPartition(
+        serviceId: ServiceID,
+        partitionId: PartitionID,
+        containerCreationConfig: ContainerCreationConfig,
+        generateRunConfigFunc: (ipAddr: string, generatedFileFilepaths: Map<string, string>, staticFileFilepaths: Map<StaticFileID, string>) => [ContainerRunConfig, Error],
+    ): [ServiceContext, Map<string, PortBinding>, Error] {
+
+        //ctx := context.Background()
+        const log: logger.Logger = new logger.Logger();
+
+        log.trace("Registering new service ID with Kurtosis API...");
+        const registerServiceArgs: RegisterServiceArgs = newRegisterServiceArgs(serviceId, partitionId);
+        // TODO TODO TODO - CALLBACK & ERROR-HANDLING
+        // registerServiceResp, err := networkCtx.client.RegisterService(ctx, registerServiceArgs)
+        // if err != nil {
+        //     return nil, nil, stacktrace.Propagate(
+        //         err,
+        //         "An error occurred registering service with ID '%v' with the Kurtosis API",
+        //         serviceId)
+        // }
+        let registerServiceResp: RegisterServiceResponse; //TODO - remove
+        const serviceIpAddr = registerServiceResp.getIpAddr();
+
+        const serviceContext: ServiceContext = ServiceContext(
+            this.client,
+            serviceId,
+            serviceIpAddr,
+            this.enclaveDataVolMountpoint,
+            containerCreationConfig.GetKurtosisVolumeMountpoint());
+        log.trace("New service successfully registered with Kurtosis API");
+
+        log.trace("Loading static files into new service namespace...");
+        const usedStaticFiles = containerCreationConfig.GetUsedStaticFiles();
+        var [staticFileAbsFilepathsOnService, err] = serviceContext.LoadStaticFiles(usedStaticFiles);
+        if (err != null) {
+            return [ null, null, err ]; //TODO - no personalized message
+        }
+        log.trace("Successfully loaded static files");
+
+        log.trace("Initializing generated files...");
+        const filesToGenerate: Map<string, boolean> = new Map();
+        for (let fileId in containerCreationConfig.GetFileGeneratingFuncs()) {
+            filesToGenerate[fileId] = true;
+        }
+        var [generatedFileFilepaths, err] = serviceContext.GenerateFiles(filesToGenerate);
+        if (err != null) {
+            return [null, null, err ] //TODO - no personalized message
+        }
+        const generatedFileAbsFilepathsOnService: Map<string, string> = new Map();
+        for (let fileId in containerCreationConfig.GetFileGeneratingFuncs()) {
```

Comment:
rather than iterating over the keys and then needing to make the first line of the for-loop be theMap[theKey], you can use the `entries()` method of a `Map` to get a `[key, value]` pair just like Go's `range` over a map. I'd recommend using this here and everywhere, in all your PRs